core/src/main/scala/org/broadinstitute/dsde/rawls/expressions/parser/antlr/LocalOutputExpressionValidationVisitor.scala

```diff
@@ -1,28 +1,67 @@
 package org.broadinstitute.dsde.rawls.expressions.parser.antlr
 
-import org.broadinstitute.dsde.rawls.RawlsException
+import akka.http.scaladsl.model.StatusCodes
 import org.broadinstitute.dsde.rawls.expressions.parser.antlr.TerraExpressionParser.{ArrContext, LiteralContext, ObjContext, RelationContext}
+import org.broadinstitute.dsde.rawls.expressions.{BoundOutputExpression, OutputExpression, ThisEntityTarget, UnboundOutputExpression, WorkspaceTarget}
+import org.broadinstitute.dsde.rawls.model.{Attributable, Attribute, AttributeNull, ErrorReport, ErrorReportSource}
+import org.broadinstitute.dsde.rawls.{RawlsException, RawlsExceptionWithErrorReport, StringValidationUtils}
 
-import scala.util.{Failure, Try}
+import scala.util.{Failure, Success, Try}
 
 /** Output expressions don't allow for entity references (relations) in the middle or for any JSON. They must
   * be of the form this.attribute or workspace.attribute */
-class LocalOutputExpressionValidationVisitor(allowRootEntity: Boolean)
-  extends LocalInputExpressionValidationVisitor(allowRootEntity = allowRootEntity) {
+class LocalOutputExpressionValidationVisitor(rootEntityTypeOption: Option[String])
+  extends TerraExpressionBaseVisitor[Try[Attribute => OutputExpression]]
```

Comment:
this no longer extends `LocalInputExpressionValidationVisitor` because the new code flow in this class is cleaner if the single inherited check is reimplemented here